eng/pipelines/templates/stages/archetype-sdk-publish-js.yml

```diff
@@ -0,0 +1,74 @@
+parameters:
+- name: BuildStages
+  type: object
+  default: []
+- name: BuildStageName
+  type: string
+  default: ''
+- name: ArtifactName
+  type: string
+  default: ''
+
+
+extends:
+  template: /eng/pipelines/templates/stages/1es-redirect.yml
+  parameters:
+    stages:
+      - ${{ parameters.BuildStages }}
+
+      # only include if running on `internal` build with manual queue, otherwise never include
+      - ${{ if and(in(variables['Build.Reason'], 'Manual', ''), eq(variables['System.TeamProject'], 'internal'))}}:
+        - stage: Publish
+          displayName: Publish
+          dependsOn: ${{ parameters.BuildStageName }}
+
+          jobs:
+            - deployment: Publish
+              environment: 'package-publish'
+              pool:
+                name: azsdk-pool-mms-ubuntu-2004-general
+                image: azsdk-pool-mms-ubuntu-2004-1espt
+                os: linux
+
+              strategy:
+                runOnce:
+                  deploy:
+                    steps:
+                    - checkout: self
+
+                    - download: current
+                      artifact: ${{ parameters.ArtifactName }}
+                      timeoutInMinutes: 5
+
+                    - task: PowerShell@2
+                      inputs:
+                        filePath: '$(Build.SourcesDirectory)/eng/scripts/determine-release-tag.ps1'
+                        failOnStderr: true
+                        pwsh: true
+
+                    - pwsh: |
+                        Write-Host "Will deploy with tag of $(Tag)"
+                        Get-ChildItem "$(Pipeline.Workspace)/drop" -Recurse -Force `
+                          | Where-Object { $_.Name -like "*.tgz" } `
+                          | Copy-Item -Destination "$(Build.ArtifactStagingDirectory)"
+
+                        Get-ChildItem "$(Build.ArtifactStagingDirectory)" -Recurse -Force | % { Write-Host $_.FullName }
+                      displayName: Move artifact to $(Build.ArtifactStagingDirectory)
+
+                    - task: EsrpRelease@7
```

Comment:
Thank you for cleaning these up and converting them to ESRP. FYI @ckairen as we will eventually want common publish template here that can be used outside of the tools repo as well.